controllers/rabbitmqcluster_controller.go

```diff
@@ -113,6 +114,20 @@ func (r *RabbitmqClusterReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, nil
 	}
 
+	if rabbitmqCluster.Spec.Image == "" {
+		rabbitmqCluster.Spec.Image = r.DefaultRabbitmqImage
+		if err = r.Update(ctx, rabbitmqCluster); err != nil {
+			if k8serrors.IsConflict(err) {
+				logger.Info("failed to update image because of conflict; requeueing...",
+					"namespace", rabbitmqCluster.Namespace,
+					"name", rabbitmqCluster.Name)
+				return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
+			}
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{RequeueAfter: 2 * time.Second}, err
```

Comment:
If the image in the spec is empty, we set it to the default, update is successful, what's the reasoning to requeue here? 🤔